CHAOS-6966 Pod JVM method latency (#8793)
harness/developer-hub: docs/chaos-engineering/use-harness-ce/chaos-faults/kubernetes/pod/pod-jvm-method-latency.md

---
id: pod-jvm-method-latency
title: Pod JVM method latency
---

Pod JVM method latency slows down the Java application executing on Kubernetes pod by introducing delays in executing the method calls.

:::tip
JVM chaos faults use the [Byteman utility](https://byteman.jboss.org/) to inject chaos faults into the JVM.
:::

![Pod JVM method latency](./static/images/pod-jvm-method-latency.png)

## Use cases
Pod JVM method latency:
- Determines the performance bottlenecks of the application.
- Tests the system's ability to handle heavy payloads.
- Evaluates the application's behavior in high-stress cases.
- Determines how quickly an application returns to normalcy after the delay.
- Determines the performance and resilience of the dependant application (or services) running on Kubernetes.

### Mandatory tunables
<table>
  <tr>
    <th> Tunable </th>
    <th> Description </th>
    <th> Notes </th>
  </tr>
  <tr>
    <td> CLASS </td>

>     <td> Specify as <b>packageName.className</b> that specifies the class in which you define the exception. </td>

Reviewer: Please check in other faults too.

```suggestion
    <td> Specify as <b>packageName.className</b> that specifies the class in which you define the target method. </td>
```

    <td> Specify it in the format `packageName.className`. For example, <code>org.framework.appName.system.WelcomeController</code>.</td>
  </tr>
  <tr>
    <td> LATENCY </td>
    <td> The delay you want to introduce in the application (in ms). </td>
    <td> Default: 2000 ms. </td>
  </tr>
  <tr>
    <td> METHOD </td>

>     <td> The method to which exception is applied. </td>

Reviewer: ```suggestion
    <td> The method in which latency is injected. </td>
```
